### PR TITLE
Remove tailwind

### PR DESCRIPTION
**Pull request by exacs**

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (me97esn) [inline .prettierrc:1]: Would prefer to remove prettier, but that's probably another PR

File: Dockerfile [hunk 2]
```diff
@@ -1,31 +1,43 @@
...
-FROM node:14 AS frontend
-WORKDIR /usr/src/app/frontend
 
-COPY ["frontend/package.json", "package.json"]
-COPY ["frontend/package-lock.json", "package-lock.json"]
-
-# See: https://stackoverflow.com/questions/18136746/npm-install-failed-with-cannot-run-in-wd
+#
+# First, we build the frontend in /usr/src/app/frontend
+FROM node:20 AS frontend
```
Comment (me97esn) [inline Dockerfile:7]: :clap:

File: backend/src/server.ts [hunk 3]
```diff
@@ -31,14 +31,13 @@ server.set("trust proxy", 1);
 server.use(
   session({
     name: "scanned-exams.sid",
+    proxy: true,
     cookie: {
-      domain: "kth.se",
+      domain: new URL(process.env.SERVER_HOST_URL || "").hostname,
```
Comment (me97esn) [inline backend/src/server.ts:36]: Is this a new environment variable that has to be set in prod before deploy?

File: backend/src/server.ts [hunk 4]
```diff
@@ -31,14 +31,13 @@ server.set("trust proxy", 1);
...
       maxAge: COOKIE_MAX_AGE_SECONDS * 1000,
       httpOnly: true,
-      secure: true,
-      sameSite: process.env.CANVAS_API_URL.endsWith("kth.se")
-        ? "strict"
-        : "none",
+      secure: "auto",
```
Comment (me97esn) [inline backend/src/server.ts:39]: Can you motivate this change? You probably know why this change is good, but I don't. I'd like to learn

File: frontend/package.json [hunk 5]
```diff
@@ -5,53 +5,38 @@
   "description": "Frontend side of the Scanned Exams service",
   "author": "KTH Royal Institute of Technology <vc.developers@kth.se> (https:kth.se)",
   "license": "MIT",
+  "source": [
+    "src/index.html"
+  ],
```
Comment (me97esn) [inline frontend/package.json:10]: What's this? I have never seen this before? How is this `source` used?

File: frontend/package.json [hunk 6]
```diff
@@ -5,53 +5,38 @@
...
     "web-vitals": "^1.1.1"
   },
   "scripts": {
-    "start": "DISABLE_ESLINT_PLUGIN=true craco start",
-    "build": "DISABLE_ESLINT_PLUGIN=true craco build",
-    "eject": "react-scripts eject",
+    "dev": "parcel --https --port 4443 --public-url /scanned-exams --host localdev.kth.se",
+    "build": "parcel build --public-url /scanned-exams --no-cache",
     "format": "prettier --write .",
     "test:lint": "eslint src/**/*.js",
```
Comment (me97esn) [inline frontend/package.json:23]: Would really prefer if we remove eslint.
But that is probably another PR

File: frontend/package.json [hunk 7]
```diff
@@ -5,53 +5,38 @@
...
     "test:format": "prettier --check .",
-    "test": "npm run test:lint && npm run test:format && react-scripts test"
+    "test": "npm run test:lint && npm run test:format"
```
Comment (me97esn) [inline frontend/package.json:25]: We have some tests in the backend, but it looks to me that they aren't run?

File: frontend/package.json [hunk 8]
```diff
@@ -5,53 +5,38 @@
...
-      "not dead",
-      "not op_mini all"
-    ],
-    "development": [
-      "last 1 chrome version",
-      "last 1 firefox version",
-      "last 1 safari version"
-    ]
-  },
-  "homepage": "/scanned-exams/app",
```
Comment (me97esn) [inline frontend/package.json:43]: Nice removal, now I don't have to ask what this means!

File: frontend/src/features/import-exams/ImportFlow.tsx [hunk 9]
```diff
@@ -47,29 +47,10 @@ export default function ImportScreen({ courseId }: any) {
...
-            <Step index={1} done={issuesDone}>
-              <StepText short="Step 2" long="2. Resolve Issues" />
-            </Step>
-            <Step index={2} done={verifyDone}>
-              <StepText short="Step 3" long="3. Verify Result" />
-            </Step>
-          </StepList>
-        </div>
+    <div>
+      <div>
```
Comment (me97esn) [inline frontend/src/features/import-exams/ImportFlow.tsx:53]: Is this really enough? I must remember to verify this when testing, because I don't see the `stepIndex` anymore?

File: frontend/src/features/import-exams/steps/PrepareImport.tsx [hunk 10]
```diff
@@ -1,13 +1,18 @@
 import React from "react";
 import { useCourseExams, useMutateImportStart } from "../../../common/api";
-import { H2, LoadingPage, PrimaryButton, P } from "../../widgets";
```
Comment (me97esn) [inline frontend/src/features/import-exams/steps/PrepareImport.tsx:3]: I had not seen this before, but _definitely agree with this removal! Was this standard html tags, but made with React? That smells bad imo

File: frontend/src/features/setup-course/SetupFlow.tsx [hunk 11]
```diff
@@ -41,33 +32,21 @@ export default function SetupScreen({
...
-            courseId={courseId}
-            onDone={() => setHomepageCreated(true)}
-          />
-        )}
-        {currentStep === 1 && <PublishCourse courseId={courseId} />}
-        {currentStep === 2 && <CreateAssignment courseId={courseId} />}
-        {currentStep === 3 && <PublishAssignment courseId={courseId} />}
-      </div>
+      <ol className={classes.Stepper}>
+        <li aria-current={currentStep === 0}>Prepare homepage</li>
```
Comment (me97esn) [inline frontend/src/features/setup-course/SetupFlow.tsx:36]: Doing this as a styled ordered list, smart! I like!

File: frontend/tsconfig.json [hunk 12]
```diff
@@ -1,26 +1,31 @@
...
+    "target": "es2016" /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */,
+
+    /* Modules */
+    "module": "commonjs" /* Specify what module code is generated. */,
+    "rootDir": "./src" /* Specify the root folder within your source files. */,
+
+    /* Emit */
+    "declaration": true /* Generate .d.ts files from TypeScript and JavaScript files in your project. */,
+    "declarationMap": true /* Create sourcemaps for d.ts files. */,
+    "sourceMap": true /* Create source map files for emitted JavaScript files. */,
```
Comment (me97esn) [inline frontend/tsconfig.json:15]: Source maps should only by used in dev mode, but it will cause lots of extra data if used in production (perhaps not critical since most users of this app use a desktop with fast internet. But still). Am I correct in my understanding that this will create source maps for production as well?

File: frontend/package.json (hunk 5 above)
Comment (exacs) [reply frontend/package.json:10]: This is how Parcel works. [Read more in the docs](https://parceljs.org/features/targets/#package.json%23source)

File: backend/src/server.ts (hunk 3 above)
Comment (exacs) [reply backend/src/server.ts:36]: The variable existed already. Set to "https://app.kth.se/scanned-exams" (in prod), "https://localdev.kth.se:4443/scanned-exams".

It is needed here because same-site cookies require the domain to be the same (including subdomain and port number)

File: backend/src/server.ts (hunk 4 above)
Comment (exacs) [reply backend/src/server.ts:39]: `secure: "auto"` means that the cookie is available both in HTTP and HTTPS. It is so to avoid problems with reverse proxies both locally (by Parcel) and in the cloud

File: frontend/tsconfig.json [hunk 13]
```diff
@@ -1,26 +1,31 @@
...
-      "dom.iterable",
-      "esnext"
-    ],
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+
+    /* Language and Environment */
+    "target": "es2016" /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */,
+
+    /* Modules */
+    "module": "commonjs" /* Specify what module code is generated. */,
```
[the remark this answers is not in the document]
Comment (exacs) [reply frontend/tsconfig.json:9]: Good question. I think some settings are just ignored by Parcel. The code _works_ and I didn't check if this file is correct, just copied from transfer-to-ladok

File: frontend/tsconfig.json (hunk 12 above)
Comment (exacs) [reply frontend/tsconfig.json:15]: Good question! I have to check it

File: frontend/src/features/import-exams/ImportFlow.tsx (hunk 9 above)
Comment (exacs) [reply frontend/src/features/import-exams/ImportFlow.tsx:53]: Yes, there is no more "Steps" in the "Import flow" since there is actually only two steps (the first one is only waiting)

File: frontend/tsconfig.json (hunk 12 above)
Comment (exacs) [reply frontend/tsconfig.json:15]: I have double-checked. Parcel does not build source maps for production. I can delete the ignored bits of the `tsconfig.json` file
